### PR TITLE
reactjs

### LINKED ISSUE #59: improve markdown preview
see #56 

* [ ] only run on certain textareas (class=markdown)
* [ ] do not show the preview always (foldout, tabs, ...)
* [ ] only show if output is none empty
* [ ] switch from jQuery to Angular? see #55

### LINKED ISSUE #60: revert angularJs changes
since adding angularJs, some things do not work anymore. for example flash messages, scrolling to thopic and display next talks. none of us can or have time to learn angularJs, so i think we should revert it.

### LINKED ISSUE #64: URL detection might be a bit broken
If you write something like http://queues.io/) in a comment, the links includes the closing bracket.

### PR DESCRIPTION
**Pull request by DavidBadura**
## General

* [x] remove angularjs (fix #60)
* [x] replace gulp with webpack
* [x] add reactjs
* [x] hot reload
* [x] reuse styling (foundation, font-awesome)
* [x] security (jwt)
* [x] form handling / validation
* [x] Markdown rendering (fix #64)
* [x] Markdown preview (fix #59)
* [x] Navigation
* [x] Notification
* [x] Datepicker
* [x] bc urls
* [x] remove twig templates
* [x] remove symfony forms
* [x] setup
* [x] production build
* [x] 404 layout
* [x] cleanup

## Actions / Use Cases

* [x] login
* [x] logout
* [x] display talks
* [x] display calendar
* [x] display archive
* [x] list filters / facetted search
* [x] subscribe / unsubscribe Speaker
* [x] vote / unvote talks
* [x] add talk
* [x] display talk
* [x] edit talk
* [x] show user profil
* [x] edit user profil
* [x] comment talk
* [x] finish talk
* [x] notifications

### CODE REVIEW

File: app/config/security.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ security:
 
         default:
             pattern: ^/
+            anonymous: ~
```
Comment (DavidBadura) [inline app/config/security.yml]: todo: remove it!

File: install.sh [hunk 2]
```diff
@@ -20,6 +20,10 @@ sudo chown -R $webserver_user:$webserver_group var/logs var/cache
 rm -f web/index_dev.php
 rm -f web/app_dev.php
 
-echo now you need to create your database
-echo afterwards, run php bin/console migrations:migrate
+echo "generate ssh keys"
```
Comment (tolry) [inline install.sh]: add instructions to set 'talkly' as passphrase or change passphrase in config.yml accordingly?

Comment (DavidBadura) [inline install.sh]: sorry, i don't understand you :D

Comment (tolry) [inline install.sh]: What I was trying to say, replace
`echo "generate ssh keys"`
with
`echo "generate ssh keys (set passprhase to talkly, when asked)"`
or something like that.

Comment (DavidBadura) [inline install.sh]: actually it should not be queried. it is set by command `pass:talkly`

Comment (tolry) [inline install.sh]: ah, my bad, missed that in the command line - didn't even know that was possible :smile:

**Comment by DavidBadura** [comment]
rebased on master

**Comment by DavidBadura** [comment]
rebased on master

File: src/js/components/NotFound/NotFound.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+export default ({children}) => {
+    return (
+        <div style={{
+            background: '#ff0000',
+            color: '#fff'
+        }}>
+            Route not found!
```
Comment (tolry) [inline src/js/components/NotFound/NotFound.jsx]: "page not found" might be more understandable to non-tech people

Comment (DavidBadura) [inline src/js/components/NotFound/NotFound.jsx]: done

File: src/TobiasOlry/TalklyBundle/Controller/TopicController.php [hunk 4]
```diff
@@ -48,19 +49,37 @@ public function listAction()
...
+            Topic::class,
+            'json',
+            DeserializationContext::create()->setSerializeNull()
+    );
+
+        $topic->setCreatedBy($this->getUser());
+
+        dump($topic);
+
+        die;
```
Comment (smoench) [inline src/TobiasOlry/TalklyBundle/Controller/TopicController.php]: @DavidBadura :fire:

File: src/TobiasOlry/TalklyBundle/Controller/TopicController.php [hunk 5]
```diff
@@ -48,19 +49,37 @@ public function listAction()
...
+        $topic = $this->get('jms_serializer')->deserialize(
+            $request->getContent(),
+            Topic::class,
+            'json',
+            DeserializationContext::create()->setSerializeNull()
+    );
+
+        $topic->setCreatedBy($this->getUser());
+
+        dump($topic);
```
Comment (smoench) [inline src/TobiasOlry/TalklyBundle/Controller/TopicController.php]: @DavidBadura :see_no_evil:

Comment (DavidBadura) [inline src/TobiasOlry/TalklyBundle/Controller/TopicController.php]: therefore it is WIP...

File: composer.json [hunk 6]
```diff
@@ -3,21 +3,17 @@
...
     "type": "project",
     "require": {
-        "php": ">=5.4.0",
-        "symfony/symfony": "^2.7",
-        "doctrine/orm": "^2.4.8",
-        "doctrine/doctrine-bundle": "~1.4",
-        "symfony/swiftmailer-bundle": "~2.3",
-        "symfony/monolog-bundle": "~2.4",
-        "sensio/distribution-bundle": "~5.0",
+        "php": ">=5.6",
```
Comment (smoench) [inline composer.json]: `"php": "^5.6 || ^7.0"`

File: src/js/components/User/Gravatar.jsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import crypto from "crypto";
+
+const URL = 'http://www.gravatar.com/avatar/';
+
+export default class Gravatar extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.email = props.children.trim().toLowerCase();
```
Comment (DavidBadura) [inline src/js/components/User/Gravatar.jsx]: todo: fix implementation

File: src/TobiasOlry/TalklyBundle/Controller/TopicController.php [hunk 8]
```diff
@@ -53,31 +49,18 @@ public function listAction()
...
         $this->getTopicService()->add($topic);
 
-        $json = $this->get('jms_serializer')->serialize(
-            $topic,
-            'json',
-            SerializationContext::create()->enableMaxDepthChecks()
-        );
-
-        return new Response($json, 200, [
+        return new Response('', 200, [
```
Comment (smoench) [inline src/TobiasOlry/TalklyBundle/Controller/TopicController.php]: `return new JsonResponse([], 200);`?

File: src/TobiasOlry/TalklyBundle/Controller/IndexController.php [hunk 9]
```diff
@@ -5,48 +5,75 @@
...
      *
-     * @return array
+     * @return Response
      */
     public function archiveAction()
     {
         $topics = $this->getTopicRepository()->findArchivedGroupByMonth();
 
-        return ['topics' => $topics];
+        $json = $this->get('serializer')->serialize(
```
Comment (smoench) [inline src/TobiasOlry/TalklyBundle/Controller/IndexController.php]: @DavidBadura ```return $this->json($topics, 200, [], ['groups' => ['topic_list']])``` => https://github.com/symfony/symfony/blob/3.1/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L113

Comment (DavidBadura) [inline src/TobiasOlry/TalklyBundle/Controller/IndexController.php]: cool, thank you!

**Comment by DavidBadura** [comment]
Ready! \o/

**Comment by tolry** [comment]
looks good, thanks again for the hard work - if there aren't any showstoppers I would merge this and work on the facetted search afterwards

**Pull request merged**